### PR TITLE
Generate HTML code for nolinebody

### PR DESCRIPTION
**Pull request by shravan1k**
### Administrative Info
Monday Board ID: 940408300
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
- Created the script to generate HTML code for nolinebody.
- Removed sk prefixes from css classes.
- Modified entry_objects to better represent data for better HTML generation

### Testing
How did you confirm your changes worked? 
- Made objects to represent the nolinebody example.
- Used the bodynoline.py file to generate HTML using the provided objects.
- Opened the generated file in my browser.

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

| Generated Alumni Advice |
| --- |
| 
Screen Shot 2021-02-02 at 10 25 51 PM
 |

### CODE REVIEW

File: py/entry_objects.py [hunk 1]
```diff
@@ -16,11 +16,40 @@ class DetailedEntry(Entry):
...
+class Description():
+    """
+    Represents a block of text related to an entry's description.
+    Attributes:
+    1. desc (list of Text, Link, and Linebreaks objects) - Each element in the list represents one part of the description.
+    """
+    def __init__(self, desc):
+        self.desc = desc
+
+class Details():
```
Comment (eliasfang) [inline py/entry_objects.py:34]: What's the difference between Description and Details?

**Review by eliasfang — CHANGES_REQUESTED** [review]
Thanks for changing all the css selectors! Was the screenshot you provided cut off at the bottom on purpose?

File: py/bodynoline.py [hunk 2]
```diff
@@ -15,6 +16,47 @@ def _bodynoline(title, entries):
         The HTML consists of tables to orient each entry within the section.
 
     Raises:
-        None
+        TypeError - If the object type of elements within the entry aren't acceptable types.
```
Comment (wes-chen) [inline py/bodynoline.py:19]: write the checks: within body, needs to be content or image, within content, needs to be text, link, or linebreak.

also add check for title = string

File: .gitignore [hunk 3]
```diff
@@ -1,4 +1,5 @@
 .DS_Store
 *.pyc
 .vscode
-newsletter.html
\ No newline at end of file
+newsletter.html
+test_vars.py
```
[the remark this answers is not in the document]
Comment (shravan1k) [reply .gitignore:5]: It's where I keep the objects used to generate the HTML code.

File: newsletter.css [hunk 4]
```diff
@@ -58,11 +58,11 @@ a {
 
 
 /* Shravan */
```
[the remark this answers is not in the document]
Comment (shravan1k) [reply newsletter.css]: removed

File: newsletter.css [hunk 5]
```diff
@@ -123,90 +123,87 @@ a {
...
+.center{
     text-align: -webkit-center;
 }
 
-.sk-test{
-    display: inline;
-}
 
 
 /* ELIAS */
```
[the remark this answers is not in the document]
Comment (shravan1k) [reply newsletter.css]: removed

File: py/bodynoline.py [hunk 6]
```diff
@@ -17,4 +18,45 @@ def _bodynoline(title, entries):
     Raises:
         None
```
[the remark this answers is not in the document]
Comment (shravan1k) [reply py/bodynoline.py]: added

File: py/entry_objects.py (hunk 1 above)
Comment (shravan1k) [reply py/entry_objects.py]: deleted details

File: py/bodynoline.py [hunk 7]
```diff
@@ -15,6 +16,47 @@ def _bodynoline(title, entries):
...
     """
-    return p('This is the body with a no line.')
+    with table() as bodynoline:
+        with tr().add(td(cls="td-valign center")).add(div(cls="title-div")):
+            h1(title, cls="section-title center")
+            hr(cls="horiz")
+        for i in range(len(entries)):
+            entry = entries[i]
+            #Adjusts padding for the first entry of the section.
+            entry_class = "td-valign"
```
Comment (wes-chen) [inline py/bodynoline.py:28]: how about this:
```
entry_class = "td-valign entry"
if i == 0:
    entry_class = "td_valign first-entry"
```

**Review by wes-chen — CHANGES_REQUESTED** [review]
making lots of progress

File: py/bodynoline.py (hunk 2 above)
Comment (shravan1k) [reply py/bodynoline.py]: added specific TypeError scenarios

File: py/bodynoline.py (hunk 7 above)
Comment (shravan1k) [reply py/bodynoline.py]: made the changes

File: py/bodynoline.py [hunk 8]
```diff
@@ -15,6 +16,51 @@ def _bodynoline(title, entries):
...
+        with tr().add(td(cls="td-valign center")).add(div(cls="title-div")):
+            if isinstance(title, str):
+                h1(title, cls="section-title center")
+                hr(cls="horiz")
+            else:
+                raise TypeError("Title must be a string.")
+        for i in range(len(entries)):
+            entry = entries[i]
+            #Adjusts padding for the first entry of the section.
+            entry_class = "td-valign entry"
```
Comment (wes-chen) [inline py/bodynoline.py:34]: `entry_class = "td-valign" +  (i==0) ? "first-entry" : "entry"`

Comment (wes-chen) [reply py/bodynoline.py:34]: will this work?

**Review by wes-chen — APPROVED** [review]
LGTM

File: .gitignore (hunk 3 above)
[the remark this answers is not in the document]
Comment (eliasfang) [reply .gitignore:5]: Sounds good!

**Review by eliasfang — APPROVED** [review]
LGTM!

**Pull request merged**